### ISSUE TITLE
Suggestion: log when a tool will be installed from sources or from the cache

### ISSUE THREAD

**Issue by tmattio** [tarides/ocaml-platform-installer#99]
suggested by @samoht

**Comment by panglesd** [comment]
For the records, here is the problem:
```
❯ ocaml-platform
* Inferring tools version...
  -> dune is already installed
  -> dune-release will be installed as dune-release+bin+platform.1.6.2-ocaml5.0.0~alpha1
  -> merlin will be installed as merlin+bin+platform.4.6.1~5.0preview-ocaml5.0.0~alpha1
  -> ocaml-lsp-server will be installed as ocaml-lsp-server+bin+platform.1.13.2~5.0preview-ocaml5.0.0~alpha1
  -> odoc will be installed as odoc+bin+platform.2.1.1-ocaml5.0.0~alpha1
  -> ocamlformat will be installed as ocamlformat+bin+platform.0.21.0-ocaml5.0.0~alpha1
* Building the tools...
  -> Creating a sandbox...
  -> [1/1] Building ocamlformat...
* Installing tools...
  -> All tools installed. For more information on the platform tools, run `ocaml-platform --help`
* Success.
```

Here, all tools are either installed or in cache, except ocamlformat. Thus, only ocamlformat is to be built, which might be confusing as it is never said that the others are not built because they have been cached.

I see the following logs as desirable:
```
❯ ocaml-platform
* Inferring tools version...
  -> dune is already installed
  -> dune-release will be installed as dune-release+bin+platform.1.6.2-ocaml5.0.0~alpha1
  -> merlin will be installed as merlin+bin+platform.4.6.1~5.0preview-ocaml5.0.0~alpha1
  -> ocaml-lsp-server will be installed as ocaml-lsp-server+bin+platform.1.13.2~5.0preview-ocaml5.0.0~alpha1
  -> odoc will be installed as odoc+bin+platform.2.1.1-ocaml5.0.0~alpha1
  -> ocamlformat will be installed as ocamlformat+bin+platform.0.21.0-ocaml5.0.0~alpha1
* Building the tools...
  -> The following tool has to be built: ocamlformat. All other tools are either already installed or found in the ocaml-platform cache.
  -> Creating a sandbox...
  -> [1/5] Building ocamlformat...
  -> [2/5] A dune-release build has been found in the cache.
  -> [3/5] A merlin build has been found in the cache. 
  -> [4/5] An ocaml-lsp-server build has been found in the cache. 
  -> [5/5] An odoc build has been found in the cache. 
* Installing tools...
  -> All tools installed. For more information on the platform tools, run `ocaml-platform --help`
* Success.
```
Before I implement this, do you think it is sufficient?

**Comment by tmattio** [comment]
That looks nice!

One alternative I was thinking about was:
```
❯ ocaml-platform
* Inferring tools version...
  -> dune is already installed
  -> dune-release will be installed from the cache as dune-release+bin+platform.1.6.2-ocaml5.0.0~alpha1
  -> merlin will be installed from the cache as merlin+bin+platform.4.6.1~5.0preview-ocaml5.0.0~alpha1
  -> ocaml-lsp-server will be installed from the cache as ocaml-lsp-server+bin+platform.1.13.2~5.0preview-ocaml5.0.0~alpha1
  -> odoc will be installed from the cache as odoc+bin+platform.2.1.1-ocaml5.0.0~alpha1
  -> ocamlformat will be installed from sources as ocamlformat+bin+platform.0.21.0-ocaml5.0.0~alpha1
* Building the tools...
  -> Creating a sandbox...
  -> [1/1] Building ocamlformat...
* Installing tools...
  -> All tools installed. For more information on the platform tools, run `ocaml-platform --help`
* Success.
```

I don't have a strong opinion though.

**Comment by panglesd** [comment]
Closed by #103.
